### PR TITLE
fix: removed plain chat from app.cal.com/video

### PR DESCRIPTION
**Pull request by nizzyabi**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

Make sure to add: 
NEXT_PUBLIC_PLAIN_CHAT_EXCLUDED_PATHS="/video"

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by Udit-takkar — COMMENTED** [review]
Changes looks fine. could you DM me the plain.com .env vars to test it locally?

**Review by PeerRich — DISMISSED** [review]

File: apps/web/lib/plain/plainChat.tsx [hunk 1]
```diff
@@ -69,26 +70,37 @@ interface PlainChatConfig {
 }
 
 const PlainChat = () => {
+  if (typeof window === "undefined") return null;
+
```
Comment (Udit-takkar) [inline apps/web/lib/plain/plainChat.tsx:74]: You shouldn't use this

Comment (Udit-takkar) [reply apps/web/lib/plain/plainChat.tsx:74]: React docs mention that 🔴 Do not call Hooks after a conditional return statement  here https://react.dev/warnings/invalid-hook-call-warning

**Review by Udit-takkar — DISMISSED** [review]
This issue could lead to errors/warning https://github.com/calcom/cal.com/pull/18385#discussion_r1898604068 because React uses a call stack to associate each hook invocation with its corresponding component instance. If a hook is skipped because a conditional return is encountered, React's internal bookkeeping becomes mismatched, causing errors.

File: apps/web/lib/plain/plainChat.tsx (hunk 1 above)
Comment (emrysal) [reply apps/web/lib/plain/plainChat.tsx:74]: oh yes hooks should be first, this should be below L79

**Comment by nizzyabi** [comment]
okay, should i use a react usestate instead or put it somewhere else?

**Comment by nizzyabi** [comment]
`const isAppDomain =
    typeof window !== "undefined" &&
    window.location.origin === process.env.NEXT_PUBLIC_WEBAPP_URL &&
    !restrictedPaths.some((path) => pathname?.startsWith(path.trim()));`

**Review by Udit-takkar — DISMISSED** [review]

File: apps/web/lib/plain/plainChat.tsx [hunk 2]
```diff
@@ -75,157 +75,179 @@ const PlainChat = () => {
...
+        }, 100);
+        return () => clearTimeout(timer);
       }
-    };
+    } catch (error) {
+      console.error("Failed to initialize Plain Chat:", error);
+    }
+  }, [userEmail, shouldOpenPlain]);
+
+  console.log("isAppDomain", isAppDomain);
```
Comment (emrysal) [inline apps/web/lib/plain/plainChat.tsx:240]: Woops 👍

**Review by emrysal — CHANGES_REQUESTED** [review]

**Review by emrysal — APPROVED** [review]
gtg

**Pull request merged**